core/src/main/java/com/coderanch/blackjack/MiniGame.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * Copyright (C) 2018 Coderanch.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.coderanch.blackjack;
+
+import com.coderanch.util.cli.InputUtility;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.security.SecureRandom;
+import java.util.ArrayDeque;
+import java.util.Deque;
+import java.util.Locale;
+import java.util.function.Predicate;
+
+import static com.coderanch.util.require.Require.requireThat;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.notNullValue;
+
+/**
+ * Mini game of Blackjack.
+ */
+public final class MiniGame {
+
+    /**
+     * Highest score in Blackjack.
+     */
+    private static final int MAX_LEGAL_SCORE = 21;
+
+    /**
+     * The predicate used for validating user input.
+     */
+    private static final Predicate<? super String> PLAY_PREDICATE = InputUtility.oneOfTheseIgnoringCase("hit", "pass");
+
+    /**
+     * The possible choices the user can make.
+     */
+    private enum CHOICE {
+        HIT, PASS
+    }
+
+    /**
+     * The player's current hand.
+     */
+    private Hand hand;
+
+    /**
+     * The deck of cards used in the game.
+     */
+    private final Deque<Card> deck = new ArrayDeque<>(Cards.getShuffledStandardDeck(new SecureRandom()));
```

Comment:
Don't create services in classes that depend on those services. Inject them into a constructor.

```java
this.inputUtility = requireThat("inputUtility", inputUtility, is(notNullValue()));
this.printStream  = requireThat("printStream",  printStream,  is(notNullValue()));

requireThat("generator", generator, is(notNullValue()));

this.deck = new ArrayDeque<>(Cards.getShuffledStandardDeck(generator));
```
Another thing, why are you using `SecureRandom` and not just `Random`?

Reply:
I did that in reaction to something you said before. I don't know what it was.

Reply:
https://github.com/CoderanchCorral/Blackjack/pull/32#discussion_r597488245

According to our previous discussion, it was Sonar that flagged this issue. If the warning appears again, I will disable it because the decision to use `Random` or `SecureRandom` is one that should be made by a developer, not an automated tool.